events/tests/test_api.py

```diff
@@ -229,3 +232,217 @@ def test_event_list_with_publisher_filters(self):
+            id='ds:place-1',
+            name='place-1',
+            data_source=self.data_source,
+            publisher=self.org_1,
+            position=Point(1, 1),
+        )
+        location_id = reverse('place-detail', kwargs={'pk': place.id})
+
+        url = reverse('event-detail', kwargs={'pk': self.event_3.id})
+        data = minimal_event_dict(self.data_source, self.org_1, location_id)
+        data['publication_status'] = 'public'
+        self.client.force_authenticate(self.user)
+        response = self.client.put(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.event_3.refresh_from_db()
+        self.assertEqual(self.event_3.publication_status, PublicationStatus.PUBLIC)
+
+    def test_admin_delete_affiliated_organization_event(self):
+        self.org_1.admin_users.add(self.user)
+        org = Organization.objects.create(
+            name='affiliated-org',
+            origin_id='affiliated-org',
+            data_source=self.data_source,
+            responsible_organization=self.org_1
+        )
+        event = Event.objects.create(
+            id='event',
+            name='event',
+            data_source=self.data_source,
+            publisher=org,
+            publication_status=PublicationStatus.DRAFT
+        )
+
+        url = reverse('event-detail', kwargs={'pk': event.id})
+        self.client.force_authenticate(self.user)
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_admin_update_affiliated_organization_event(self):
+        self.org_1.admin_users.add(self.user)
+        org = Organization.objects.create(
+            name='affiliated-org',
+            origin_id='affiliated-org',
+            data_source=self.data_source,
+            responsible_organization=self.org_1
+        )
+        event = Event.objects.create(
+            id='event',
+            name='event',
+            data_source=self.data_source,
+            publisher=org,
+            publication_status=PublicationStatus.DRAFT
+        )
+        place = Place.objects.create(
+            id='ds:place-1',
+            name='place-1',
+            data_source=self.data_source,
+            publisher=self.org_1,
+            position=Point(1, 1),
+        )
+        location_id = reverse('place-detail', kwargs={'pk': place.id})
+
+        url = reverse('event-detail', kwargs={'pk': event.id})
+        data = minimal_event_dict(self.data_source, self.org_1, location_id)
+        data['publication_status'] = 'public'
+        self.client.force_authenticate(self.user)
+        response = self.client.put(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        event.refresh_from_db()
+        self.assertEqual(event.publication_status, PublicationStatus.PUBLIC)
+
+    def test_regular_user_create_public_event_denied(self):
```

Comment:
I guess for completeness' sake, we should add the same tests for a regular user trying to PUT or DELETE any public events?

Reply:
Make sense